app/controllers/actions/ChannelUtil.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.actions
+
+import models.ChannelType
+import models.ChannelType.{api, web}
+import play.api.mvc.Request
+
+private[actions] object ChannelUtil {
```

Comment:
This is only ever used by ChannelAction; could the logic just be moved there?  And if ChannelType had a way to be constructed from a string (e.g. an apply method that returns a Try[ChannelType] or Option[ChannelType], you wouldn't need to do all the matching against string representations here